backends/rapidpro/backend.go

```diff
@@ -296,6 +296,23 @@ func (b *backend) WriteChannelLogs(ctx context.Context, logs []*courier.ChannelL
 	return nil
 }
 
+// Check if external ID has been seen in a period
+func (b *backend) CheckExternalIDSeen(msg courier.Msg) courier.Msg {
+	var prevUUID = checkExternalIDSeen(b, msg)
+	m := msg.(*DBMsg)
```

Comment:
is this conversion needed?

Reply:
Yes, as far as I understand, `alreadyWritten` is not defined on `courier.Msg`, only on `DBMsg`.

Reply:
Okay, actually that is fine since `DBMsg` implements the interface we are returning `courier.Msg`